### PR TITLE
Refactor fn to closure

### PR DESCRIPTION
**Pull request by HaoranYi**
#### Problem
Small refactor to use closure instead of fn for merge hashmap counters.

#### Summary of Changes

Fixes #

### CODE REVIEW

**Review by sakridge — COMMENTED** [review]
What's the improvement?

File: core/src/retransmit_stage.rs [hunk 1]
```diff
@@ -314,19 +314,22 @@ fn retransmit(
             .fetch_add(retransmit_time.as_us(), Ordering::Relaxed);
         num_nodes
     };
-    fn merge<K, V>(mut acc: HashMap<K, V>, other: HashMap<K, V>) -> HashMap<K, V>
```
Comment (behzadnouri) [inline core/src/retransmit_stage.rs:317]: I personally think this original form is more readable.
The function body is also shorter.

**Comment by sakridge** [comment]
And the test is testing a copy of the logic?

**Comment by HaoranYi** [comment]
Functionally no changes. I just don't like the nested function and prefer closures. 
Yes. the test is to test the merge closure with copied code.

File: core/src/retransmit_stage.rs (hunk 1 above)
Comment (HaoranYi) [reply core/src/retransmit_stage.rs:317]: One downside of the original form is that you have to specify all the generic parameters and traits, while for closure, the compiler can infer it.

**Comment by HaoranYi** [comment]
> And the test is testing a copy of the logic?

Yes, it is. I could extract lambda out and avoid the copy.

**Comment by HaoranYi** [comment]
Since there is no functional change, and using closure or function are just personal preference. 
We will keep the original code. 
Close this pull request. :)

**Pull request closed without merging**